dfget/core/downloader/p2p_downloader.go

```diff
@@ -203,18 +205,46 @@ func (p2p *P2PDownloader) pullPieceTask(item *Piece) (
 	return res, err
 }
 
-func (p2p *P2PDownloader) pullRate(data *types.PullPieceTaskResponseContinueData) {
+// getPullRate get download rate limit dynamically.
+func (p2p *P2PDownloader) getPullRate(data *types.PullPieceTaskResponseContinueData) int {
+	var localRate int
 
+	if p2p.Cfg.LocalLimit > 0 {
+		localRate = p2p.Cfg.LocalLimit
+	} else {
+		localRate = data.DownLink * 1024
+	}
+
+	// Calculate the download speed limit
+	// that the current download task can be assigned
+	// by the uploader server.
+	// TODO: Reduce the frequency of requests.
+	url := fmt.Sprintf("http://%s:%d%s%s", p2p.Cfg.RV.LocalIP, p2p.Cfg.RV.PeerPort, config.LocalHTTPPathRate, p2p.taskFileName)
```

Comment:
How about to create an interface of dfget http server just like `SupernodeAPI`? It makes the code more readable and reusable.

Reply:
May be do that with the later pull request? 
At the same time, I also want to define some constants  about http headers, etc.

Reply:
ok